src/core/Button/style.ts

```diff
@@ -1,23 +1,40 @@
 import styled from "@emotion/styled";
-import { Button } from "@material-ui/core";
+import { Button, ButtonProps } from "@material-ui/core";
 import {
   fontCapsXxxs,
+  getBorders,
   getColors,
   getCorners,
   getSpaces,
   Props,
 } from "../styles";
 
-const sdsPropNames = ["isAllCaps", "isRounded", "sdsStyle", "sdsType"];
+export interface ExtraProps {
+  color?: "primary" | "error";
```

Comment:
@tihuan I changed this to only accept the colors that we have used / defined for buttons.  The problem I was running into was that some of the colors aren't defined in our theme at all, or they're missing certain various (`400`, `600`, etc.).  

However, this brings up bit of a philosophical question here.  Are we ok with just defining components that we need as we go (essentially the approach taken here, since I'm mainly working on this to add the error style), which keeps the SDS to the minimal set of components that we actually use?   Or do we want to flesh out all the possible color values for buttons that could be used in MUI so that they will be available in the future if someone needed them (so this would involve having the whole list of color and making sure they work)?

Reply:
Hey Jerry! I want to jump in here—our philosophy with the SDS is for it to be opinionated and specific, meaning we only plan on adding components and their variants as we see the need arise, this is what helps distinguish our design system from simply becoming a sprawling component library. There are a few instances where we have deviated from that (for example anticipating the need for a primary square dropdown button, even though none of our products currently use it, it seemed like there was enough of a potential need to just create it), but overall our approach is to define components that we need as we go. Lmk if you've got any questions!

Reply:
Thanks for clarifying @clarsen-czi!  That makes sense and seems like the right approach to me, to just add what we need as we go instead of defining a broad set of components that may never be used.

Reply:
Hey all! Sorry for the late reply 😆 On PTO today!

Yeah I think limiting the color value scope sounds great and matches the actual behavior 👍 

Thanks for updating that!

Reply:
@tihuan Get out of here! You're on PTO! 🤣